features/22-email-verification/api.md

```diff
@@ -0,0 +1,111 @@
+# Verify by email
+The use case and requirement apply to verify-by-email and verify-by-sms.
+
+## Use cases
+* App may request the user to verify the email address/phone number upon signup.
+* The app need will send an email/SMS to the user with the email address/phone number provided when logging in. The app will want to customize the content of the email/SMS.
+* Both email/SMS verification supports verify by a verification code or by a verification link.
+	* For verification code, the user will type the verification code from the email address/SMS an enter it to the app to complete verification.
+	* For verification link, the user will be redirected to a webpage, which automatically send the code to the server to complete verification.
+* The App is able to check the verification status of the user using the SDK and cloud function so that it can display other info and/or disable features.
+* The developer may want to deny unverified user from accessing any server API except auth related ones.
+
+## Requirements
+* Server will have configuration options for:
+	* enable/disable email verification (boolean)
+	* enable/disable SMS verification (boolean)
+	* set verification as required/optional (boolean)
+	* smtp server settings
+	* SMS gateway settings
+	* email template
+	* SMS template
```

Comment:
I guess we will need a web template (kind of like forget-password) after user click the verification links? It could be a web template or a redirection URL (if anything could be done to make deep-linking in app easier, should be included too)